### PR TITLE
feat(s3 source): Add `delete_failed_message` configuration option

### PR DESCRIPTION
**Pull request by tanushri-sundar**
Addresses #19711

### CODE REVIEW

File: website/cue/reference/components/sources/base/aws_s3.cue [hunk 1]
```diff
@@ -524,6 +524,15 @@ base: components: sources: aws_s3: configuration: {
 					unit: "tasks"
 				}
 			}
+			delete_failed_message: {
+				description: """
+					Whether to delete non-retryable messages
```
Comment (maycmlee) [inline website/cue/reference/components/sources/base/aws_s3.cue:529]:
```suggestion
					Whether to delete non-retryable messages.
```

File: website/cue/reference/components/sources/base/aws_s3.cue [hunk 2]
```diff
@@ -524,6 +524,15 @@ base: components: sources: aws_s3: configuration: {
...
+
+					If a message is rejected by the sink and not retryable, it will be deleted from the queue.
```
Comment (maycmlee) [inline website/cue/reference/components/sources/base/aws_s3.cue:531]:
```suggestion
					If a message is rejected by the sink and not retryable, it is deleted from the queue.
```

**Review by maycmlee — COMMENTED** [review]
Left a couple of suggestions

File: src/sources/aws_s3/sqs.rs [hunk 3]
```diff
@@ -102,6 +102,13 @@ pub(super) struct Config {
     #[derivative(Default(value = "default_true()"))]
     pub(super) delete_message: bool,
 
+    /// Whether to delete non-retryable messages
+    ///
+    /// If a message is rejected by the sink and not retryable, it will be deleted from the queue.
```
Comment (maycmlee) [inline src/sources/aws_s3/sqs.rs:107]:
```suggestion
    /// Whether to delete non-retryable messages.
    ///
    /// If a message is rejected by the sink and not retryable, it is deleted from the queue.
```

**Review by maycmlee — COMMENTED** [review]
Sorry, I think the change needed to be made in the code comment

**Review by maycmlee — APPROVED** [review]
👍 for docs

**Review by jszwedko — COMMENTED** [review]
Thanks for contributing this @tanushri-sundar ! The implementation looks good to me; I appreciate you added an option to maintain compatibility.

I'll kick off the integration tests to make sure they still pass. I do see that we have a test for rejections already though it would only test the current behavior. If you feel inspired, a test testing the opt-in behavior would be nice too.

Other than that I think you just need a changelog fragment describing the change. See https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md for details on this.

**Comment by tanushri-sundar** [comment]
@jszwedko I added an integration test and tried to trigger `/ci-run-integration-aws`, but it seems like a Vector code owner has to do it. 

I think the tests should pass, but I faced some issues in my local environment, so I'd like to make sure it works in the standard cloud test environment.

**Comment by jszwedko** [comment]
Thanks @tanushri-sundar ! I'll rerun the integration tests (you are correct that a maintainer needs to execute that workflow).

**Comment by tanushri-sundar** [comment]
Thanks @jszwedko ! It looks like some of the checks are still pending - is that expected?

**Comment by tanushri-sundar** [comment]
Are there constraints on what the author name in a changelog fragment can be? I see the `Check Spelling` test failing with:

```
Error: changelog.d/19711_add_s3_source_delete_configuration.feature.md:3:11 ... 19, Error - `tanushri` is not a recognized word. (unrecognized-spelling)
Error: changelog.d/19711_add_s3_source_delete_configuration.feature.md:3:20 ... 26, Error - `sundar` is not a recognized word. (unrecognized-spelling)
```

**Comment by tanushri-sundar** [comment]
@jszwedko Would you mind re-triggering the workflows? I fixed the changelog and allow.txt. Apologies for the spam and thank you!

**Comment by jszwedko** [comment]
> @jszwedko Would you mind re-triggering the workflows? I fixed the changelog and allow.txt. Apologies for the spam and thank you!

No problem!

**Comment by jszwedko** [comment]
Thanks for this contribution @tanushri-sundar !

**Pull request merged**